pkgs/top-level/release.nix

```diff
@@ -15,8 +15,14 @@
 , limitedSupportedSystems ? [ "i686-linux" ]
   # Strip most of attributes when evaluating to spare memory usage
 ,  scrubJobs ? true
-  # Attributes passed to nixpkgs. Don't build packages marked as unfree.
-,  nixpkgsArgs ? { config = { allowUnfree = false; inHydra = true; }; }
+  # Attributes passed to nixpkgs. Don't build packages marked as unfree, unless
+  # they are also marked as redistributable.
+  # TODO: Consider adding this predicate to stdenv's check-meta for generic re-use?
+,  nixpkgsArgs ? { config = {
+     allowUnfree = false;
+     allowUnfreePredicate = (p: p.meta.license.redistributable or false);
+     inHydra = true;
+   }; }
```

Comment:
We could add an attribute that can go into the config called `allowUnfreeRedistributable`, that gets checked before the more generic `allowUnfree` in the evaluator:
https://github.com/NixOS/nixpkgs/blob/18.03/pkgs/stdenv/generic/check-meta.nix#L184-L185

and has a nice warning message letting you know that you could get the pkg either by setting `allowUnfree = true` or `allowUnfreeRedistributable = true` in your `config`.  I'm not entirely sure if this is the way we want to go, since at an end-user level there's not a huge difference (it's more of a Hydra/distribution distinction), and the meta evaluator is already quite complex and heavy -- so I figured I'd open a WIP PR to see if anyone else has thoughts.